src/raftstore/store/peer.rs

```diff
@@ -2123,6 +2107,10 @@ impl Peer {
     }
 
     pub fn get_peer_from_cache(&self, peer_id: u64) -> Option<metapb::Peer> {
+        if peer_id == 0 {
```

Comment:
Why add it?

Reply:
When there is no leader, `peer_id` may be 0 as it represents leader id. Add a check to skip slow path.